lib/store/inventory.ex

```diff
@@ -309,11 +309,69 @@ defmodule Store.Inventory do
       if order_product != nil do
         {:error, "Product already in cart"}
       else
-        create_order_product(%{order_id: order.id, product_id: product.id})
+        {val, _} = Integer.parse(product_params["quantity"])
+        size = product_params["size"]
+
+        if update_stock_sizes(product, size, val) == {:error, "Not enough stock"} do
+          {:error, "Not enough stock"}
+        else
+          create_order_product(%{
+            order_id: order.id,
+            product_id: product.id,
+            quantity: val,
+            size: size
+          })
+        end
       end
     else
       {:ok, order} = create_order(%{user_id: user.id})
-      create_order_product(%{order_id: order.id, product_id: product.id})
+      {val, _} = Integer.parse(product_params["quantity"])
+      size = product_params["size"]
+
+      if update_stock_sizes(product, size, val) == {:error, "Not enough stock"} do
+        {:error, "Not enough stock"}
+      else
+        create_order_product(%{
+          order_id: order.id,
+          product_id: product.id,
+          quantity: val,
+          size: size
+        })
+      end
+    end
+  end
+
+  def update_stock_sizes(product, size, quantity) do
+    new_sizes =
+      case size do
+        "XS" -> Map.put(product.sizes, :xs_size, product.sizes.xs_size - quantity)
+        "S" -> Map.put(product.sizes, :s_size, product.sizes.s_size - quantity)
+        "M" -> Map.put(product.sizes, :m_size, product.sizes.m_size - quantity)
+        "L" -> Map.put(product.sizes, :l, product.sizes.l_size - quantity)
+        "XL" -> Map.put(product.sizes, :xl_size, product.sizes.xl_size - quantity)
+        "XXL" -> Map.put(product.sizes, :xxl_size, product.sizes.xxl_size - quantity)
+      end
+
+    values = Map.values(new_sizes) |> Enum.filter(&is_integer/1)
```

Comment:
I converted the map of sizes into a list and filtered out any non-integer values, such as the "id" and "__struct__" entries in the map.